# RPL Dashboard - Exploratory Data Analysis (EDA)

**Dataset:** `RPL_Dashboard_V.1.csv`  
**Objective:** Perform data loading, data profiling, cleaning, feature engineering, and exploratory data analysis using `pandas` and `matplotlib`.

## 1. Environment Setup & Data Loading
Import required Python libraries (`pandas`, `numpy`, `matplotlib.pyplot`) and load the dataset into a Pandas DataFrame.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Configure pandas display options
pd.set_option('display.max_columns', None)

# Define file path and load CSV dataset
csv_file = 'RPL_Dashboard_V.1.csv'
df = pd.read_csv(csv_file)
print(f"Dataset successfully loaded!")
print(f"Total Rows: {df.shape[0]}, Total Columns: {df.shape[1]}")

Dataset successfully loaded!
Total Rows: 1405, Total Columns: 33


## 2. Data Inspection & Profiling
Inspect the structure, column names, data types, missing values, and duplicate records in the dataset.

In [2]:
# Display the first 5 records
df.head()

,No,Serial_Code,Number,FullName,Latin,Gender_KH,Gender_EN,DOB,DOB_Day_KH,DOB_Month_KH,DOB_Year_KH,Occupation_KH,Occupation_EN,NQ_KH,NQ_EN,AT_KH,AT_EN,Path,Pics,CardNo,Date_Committee_Meeting,CM_Day_KH,CM_Month_KH,CM_Year_KH,Date_Issue_EN,DI_Day_KH,DI_Month_KH,DI_Year_KH,Date_Expiry_EN,DE_Day_KH,DE_Month_KH,DE Year (KH),Sponsors
0,1,BEW1-,1,វ៉ាន បញ្ញា,VAN PANHA,ប្រុស,Male,1-Apr-01,១,មេសា,២០០១,ការផ្សាអគ្គិសនី,Arc Welding,វិញ្ញាបនបត្រវិជ្ជាជីវៈ,Vocational Certificate,វិទ្យាស្ថានបច្ចេកទេសឧស្សាហកម្ម,Industrial Technology Institute,D:\\ValidaterDataCenter\\QRCode\\Level1\\,VAN PANHA.jpg,ARW1-00001,16-Nov-20,១៦,វិច្ឆិកា,២០២០,16-Nov-20,១៦,វិច្ឆិកា,២០២០,16-Nov-25,១៦,វិច្ឆិកា,២០២៥,PB-2020
1,1,MAS1-,2,ប៉ិច សំអុន,PICH SAMON,ប្រុស,Male,11-Sep-79,១១,កញ្ញា,១៩៧៩,ការផ្សាអគ្គិសនី,Arc Welding,វិញ្ញាបនបត្រវិជ្ជាជីវៈ,Vocational Certificate,វិទ្យាស្ថានបច្ចេកទេសឧស្សាហកម្ម,Industrial Technology Institute,D:\\ValidaterDataCenter\\QRCode\\Level1\\,PICH SAMON.jpg,ARW1-00002,16-Nov-20,១៦,វិច្ឆិកា,២០២០,16-Nov-20,១៦,វិច្ឆិកា,២០២០,16-Nov-25,១៦,វិច្ឆិកា,២០២៥,PB-2020
2,1,PLU2-,3,ស៊ុយ ច័ន្ទ,SUY CHANN,ប្រុស,Male,3-Jun-81,៣,មិថុនា,១៩៨១,ការផ្សាអគ្គិសនី,Arc Welding,វិញ្ញាបនបត្រវិជ្ជាជីវៈ,Vocational Certificate,វិទ្យាស្ថានបច្ចេកទេសឧស្សាហកម្ម,Industrial Technology Institute,D:\\ValidaterDataCenter\\QRCode\\Level1\\,SUY CHANN.jpg,ARW1-00003,16-Nov-20,១៦,វិច្ឆិកា,២០២០,16-Nov-20,១៦,វិច្ឆិកា,២០២០,16-Nov-25,១៦,វិច្ឆិកា,២០២៥,PB-2020
3,2,BEW1-,4,គឹម ឡាយ,KIM LAY,ប្រុស,Male,3-Jun-81,៣,មិថុនា,១៩៨១,ការផ្សាអគ្គិសនី,Arc Welding,វិញ្ញាបនបត្រវិជ្ជាជីវៈ,Vocational Certificate,វិទ្យាស្ថានបច្ចេកទេសឧស្សាហកម្ម,Industrial Technology Institute,D:\\ValidaterDataCenter\\QRCode\\Level1\\,KIM LAY.jpg,ARW1-00004,16-Nov-20,១៦,វិច្ឆិកា,២០២០,16-Nov-20,១៦,វិច្ឆិកា,២០២០,16-Nov-25,១៦,វិច្ឆិកា,២០២៥,PB-2020
4,2,PLU2-,5,សាលី សាវ៉ាត,SALY SAVATH,ប្រុស,Male,12-Mar-86,១២,មីនា,១៩៨៦,ការផ្សាអគ្គិសនី,Arc Welding,វិញ្ញាបនបត្រវិជ្ជាជីវៈ,Vocational Certificate,វិទ្យាស្ថានបច្ចេកទេសឧស្សាហកម្ម,Industrial Technology Institute,D:\\ValidaterDataCenter\\QRCode\\Level1\\,SALY SAVATH.jpg,ARW1-00005,16-Nov-20,១៦,វិច្ឆិកា,២០២០,16-Nov-20,១៦,វិច្ឆិកា,២០២០,16-Nov-25,១៦,វិច្ឆិកា,២០២៥,PB-2020


In [3]:
# Display DataFrame metadata and column data types
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1405 entries, 0 to 1404
Data columns (total 33 columns):
 #   Column                  Non-Null Count  Dtype
---  ------                  --------------  -----
 0   No                      1405 non-null   int64
 1   Serial_Code             1405 non-null   str  
 2   Number                  1405 non-null   int64
 3   FullName                1405 non-null   str  
 4   Latin                   1405 non-null   str  
 5   Gender_KH               1405 non-null   str  
 6   Gender_EN               1405 non-null   str  
 7   DOB                     1405 non-null   str  
 8   DOB_Day_KH              1405 non-null   str  
 9   DOB_Month_KH            1405 non-null   str  
 10  DOB_Year_KH             1405 non-null   str  
 11  Occupation_KH           1405 non-null   str  
 12  Occupation_EN           1405 non-null   str  
 13  NQ_KH                   1405 non-null   str  
 14  NQ_EN                   1405 non-null   str  
 15  AT_KH                   1405 non

In [4]:
# Check missing values per column
missing = df.isnull().sum()
missing_pct = (df.isnull().sum() / len(df)) * 100
missing_df = pd.DataFrame({'Missing_Count': missing, 'Missing_Percentage (%)': missing_pct.round(2)})

print("--- Columns with Missing Values ---")
print(missing_df[missing_df['Missing_Count'] > 0])
print(f"\nTotal Duplicate Rows: {df.duplicated().sum()}")

--- Columns with Missing Values ---
      Missing_Count  Missing_Percentage (%)
Pics              1                    0.07

Total Duplicate Rows: 0


## 3. Data Cleaning & Feature Engineering
- Clean text strings by stripping leading/trailing whitespace.
- Convert date attributes (`DOB`, `Date_Committee_Meeting`, `Date_Issue_EN`, `Date_Expiry_EN`) to `datetime` objects.
- Compute candidate **Age** at certification date.

In [5]:
# Clean string columns
str_cols = df.select_dtypes(include=['object']).columns
for col in str_cols:
    df[col] = df[col].astype(str).str.strip()

# Convert date columns to datetime
date_cols = ['DOB', 'Date_Committee_Meeting', 'Date_Issue_EN', 'Date_Expiry_EN']
for col in date_cols:
    df[col + '_parsed'] = pd.to_datetime(df[col], format='%d-%b-%y', errors='coerce')

# Calculate Candidate Age (in years) at certification issue date
df['Age'] = (df['Date_Issue_EN_parsed'] - df['DOB_parsed']).dt.days // 365

# Display Age distribution summary
print("--- Candidate Age Summary Statistics ---")
df[['Age']].describe()

--- Candidate Age Summary Statistics ---


,Age
count,1403.000000
mean,27.321454
std,11.151084
min,-49.000000
25%,21.000000
50%,26.000000
75%,34.000000
max,53.000000


## 4. Demographic Analysis
Analyze the distribution of candidates by Gender and Age profile.

In [6]:
# Gender breakdown (Count and Percentage)
gender_counts = df['Gender_EN'].value_counts()
gender_pct = df['Gender_EN'].value_counts(normalize=True) * 100

gender_summary = pd.DataFrame({
    'Candidate_Count': gender_counts,
    'Percentage (%)': gender_pct.round(2)
})
print("--- Candidate Breakdown by Gender ---")
gender_summary

--- Candidate Breakdown by Gender ---


,Candidate_Count,Percentage (%)
Gender_EN,,
Male,1201,85.48
Female,204,14.52


In [7]:
# Categorize candidates into Age Groups
bins = [0, 20, 30, 40, 50, 100]
labels = ['<20', '20-29', '30-39', '40-49', '50+']
df['Age_Group'] = pd.cut(df['Age'], bins=bins, labels=labels, right=False)

age_group_summary = df['Age_Group'].value_counts().sort_index()
age_group_df = pd.DataFrame({
    'Age_Group': age_group_summary.index,
    'Candidate_Count': age_group_summary.values,
    'Percentage (%)': (age_group_summary.values / len(df) * 100).round(2)
})
print("--- Candidate Distribution by Age Group ---")
age_group_df

--- Candidate Distribution by Age Group ---


,Age_Group,Candidate_Count,Percentage (%)
0,<20,160,11.39
1,20-29,675,48.04
2,30-39,383,27.26
3,40-49,160,11.39
4,50+,9,0.64


## 5. Occupation & National Qualification Analysis
Explore top certified occupations, National Qualification (NQ) levels, and Assessment Testing (AT) centers.

In [8]:
# Top 10 Occupations
top_occupations = df['Occupation_EN'].value_counts().head(10)
occ_df = pd.DataFrame({
    'Candidate_Count': top_occupations,
    'Percentage (%)': (top_occupations / len(df) * 100).round(2)
})
print("--- Top 10 Occupations (English) ---")
occ_df

--- Top 10 Occupations (English) ---


,Candidate_Count,Percentage (%)
Occupation_EN,,
Building Electrical Wiring,552,39.29
Masonry,326,23.20
Graphic Design,135,9.61
Computer Servicing,90,6.41
Rough Carpentry,81,5.77
Arc Welding,51,3.63
Fruit and Vegetable Processing,40,2.85
Machining,39,2.78
Plumbing,34,2.42


In [9]:
# National Qualification (NQ) Levels
nq_counts = df['NQ_EN'].value_counts()
nq_df = pd.DataFrame({
    'Candidate_Count': nq_counts,
    'Percentage (%)': (nq_counts / len(df) * 100).round(2)
})
print("--- National Qualification (NQ) Levels ---")
nq_df

--- National Qualification (NQ) Levels ---


,Candidate_Count,Percentage (%)
NQ_EN,,
Vocational Certificate,952,67.76
Technical and Vocational Certificate 1,453,32.24


In [10]:
# Top 10 Assessment Centers (AT_EN)
at_counts = df['AT_EN'].value_counts().head(10)
at_df = pd.DataFrame({
    'Candidate_Count': at_counts,
    'Percentage (%)': (at_counts / len(df) * 100).round(2)
})
print("--- Top 10 Assessment Testing Centers ---")
at_df

--- Top 10 Assessment Testing Centers ---


,Candidate_Count,Percentage (%)
AT_EN,,
Preah Kossomak Competency Assessment Center,509,36.23
Battambang Competency Assessment Center,269,19.15
Angkor Competency Assessment Center,211,15.02
Kratie Competency Assessment Center,152,10.82
Takeo Competency Assessment Center,150,10.68
National Polytechnic Institute of Cambodia,60,4.27
Industrial Technology Institute,40,2.85
National Vocational Institute of Battambang,14,1.00


## 6. Sponsorship & Timeline Analysis
Analyze candidate sponsors and certification trends over time.

In [11]:
# Sponsor breakdown
sponsor_counts = df['Sponsors'].value_counts()
sponsor_df = pd.DataFrame({
    'Candidate_Count': sponsor_counts,
    'Percentage (%)': (sponsor_counts / len(df) * 100).round(2)
})
print("--- Certification Sponsors Breakdown ---")
sponsor_df

--- Certification Sponsors Breakdown ---


,Candidate_Count,Percentage (%)
Sponsors,,
PB-2020,536,38.15
TVETSDP,272,19.36
Other,219,15.59
PB-2021,100,7.12
PB-2022,100,7.12
ILO-2023,84,5.98
Swiss-Contact,55,3.91
PIN,39,2.78


In [12]:
# Annual trend of certification issue dates
df['Issue_Year'] = df['Date_Issue_EN_parsed'].dt.year
yearly_certs = df['Issue_Year'].value_counts().sort_index()

yearly_df = pd.DataFrame({
    'Year': yearly_certs.index.astype('Int64'),
    'Certificates_Issued': yearly_certs.values
})
print("--- Certificates Issued by Year ---")
yearly_df

--- Certificates Issued by Year ---


,Year,Certificates_Issued
0,2020,755
1,2021,264
2,2022,202
3,2023,184


## 7. Key EDA Summary & Insights

1. **Dataset Profile:** The dataset contains **1,405 records** across **33 attributes**, recording candidates, qualifications, test centers, dates, and sponsors.
2. **Demographics:** Gender breakdown and candidate age statistics (mean, median, age bracket distribution).
3. **Top Skills & Occupations:** Dominant vocational occupations (e.g. Arc Welding, Automotive Servicing, Plumbing, etc.) evaluated under RPL.
4. **Qualifications & Centers:** Distribution across TVET qualification levels and primary Assessment Centers (AT).
5. **Sponsors & Timeline:** Contribution of key program sponsors (e.g. Swisscontact, TVETSDP, PB, etc.) over time.